packages/jest-config/package.json

```diff
@@ -12,8 +12,6 @@
     "babel-jest": "^22.4.1",
     "chalk": "^2.0.1",
     "glob": "^7.1.1",
-    "jest-environment-jsdom": "^22.4.1",
-    "jest-environment-node": "^22.4.1",
```

Comment:
You cannot remove them from here, because they are required by the following line:

https://github.com/facebook/jest/blob/master/packages/jest-config/src/utils.js#L123

If you remove them from the package list, you start relying on the hoisting to make sure that the environments are siblings of `jest-config`, which is an unsafe assumption.

Reply:
Looks like a require based on a runtime variable is done in both [jest-runner](https://github.com/facebook/jest/blob/v22.4.2/packages/jest-runner/src/run_test.js#L69) and [jest-config](https://github.com/facebook/jest/blob/v22.4.2/packages/jest-config/src/utils.js#L116). I guess that has implications on @SimenB's suggestion "pass a full path through to the runner" too.

I suppose that leaves us with the option of some refactoring towards dependency injection, and maybe @pgonzal's suggestion with a require done at config time is a simple way to do that.